src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -456,27 +453,41 @@ private string CreateInitialQuery()
             string CatalogName = parts[MultipartIdentifier.CatalogIndex];
             if (isTempTable && string.IsNullOrEmpty(CatalogName))
             {
-                TDSCommand += string.Format("exec tempdb..{0} N'{1}.{2}'",
-                    TableCollationsStoredProc,
-                    SchemaName,
-                    TableName
-                );
+                CatalogName = "tempdb";
             }
-            else
+            else if (!string.IsNullOrEmpty(CatalogName))
             {
-                // Escape the catalog name
-                if (!string.IsNullOrEmpty(CatalogName))
-                {
-                    CatalogName = SqlServerEscapeHelper.EscapeIdentifier(CatalogName);
-                }
-                TDSCommand += string.Format("exec {0}..{1} N'{2}.{3}'",
-                    CatalogName,
-                    TableCollationsStoredProc,
-                    SchemaName,
-                    TableName
-                );
+                CatalogName = SqlServerEscapeHelper.EscapeIdentifier(CatalogName);
             }
-            return TDSCommand;
+
+            string objectName = ADP.BuildMultiPartName(parts);
+            string escapedObjectName = SqlServerEscapeHelper.EscapeStringAsLiteral(objectName);
+            // Specify the column names explicitly. This is to ensure that we can map to hidden columns (e.g. columns in temporal tables.)
+            // If the target table doesn't exist, OBJECT_ID will return NULL and @Column_Names will remain non-null. The subsequent SELECT *
+            // query will then continue to fail with "Invalid object name" rather than with an unusual error because the query being executed
+            // is NULL.
+            // Some hidden columns (e.g. SQL Graph columns) cannot be selected, so we need to exclude them explicitly.
+            return $"""
+SELECT @@TRANCOUNT;
+
+DECLARE @Column_Names NVARCHAR(MAX) = NULL;
+IF EXISTS (SELECT TOP 1 * FROM sys.all_columns where object_id = object_id('sys.all_columns') AND [name] = 'graph_type')
```

Comment:
I suppose since the results of this SELECT are only used to determine if anything exists, then it doesn't matter that we're using * here.